### ISSUE TITLE
Feature request: backport `typing` fix in 3.11 to evaluate strings as forward refs in PEP 585 generics

### ISSUE THREAD

**Issue by egparedes** [alexmojaki/eval_type_backport#10]
Python 3.11 also added a bug fix to transform type strings to `ForwardRefs` when evaluating generics args here:
[bpo-41370: Evaluate strings as forward refs in PEP 585 generics #30900](https://github.com/python/cpython/pull/30900) 

Would you consider adding that fix to this package? It doesn't seem like a complicated feature to add and it would help to deal with typing annotations in a more uniform way across different python versions.

**Comment by alexmojaki** [comment]
Yes, I'd accept a PR for that. Did you find this via pydantic? Are you using it somehow?

**Comment by egparedes** [comment]
Sorry for the late reply, I've been very busy lately with other issues. I've used pydantic in the past and found it awesome but in the end we had to replace it in our project for some very specific issues related to our use case. I arrived to this repo through @bswck [comments](https://github.com/python/typing_extensions/issues/315) in the typing_extensions repo, and my idea was to use it as a more stable replacement for some hacks used internally in my own projects (not using pydantic).

I'm not sure if I'll be able to work on this feature in the short term, so feel free to open your own PR if you don't want to wait.

**Comment by alexmojaki** [comment]
I got an AI to generate https://github.com/alexmojaki/eval_type_backport/pull/34. I tried looking at it, but decided that I don't have the time to fully understand it in depth the way I'd like.

It's also not clear to me that there's any significant overlap between this problem and the things this library already solves.
